internal/buffer/settings.go

```diff
@@ -8,6 +8,43 @@ import (
 	"github.com/zyedidia/micro/v2/internal/screen"
 )
 
+func (b *Buffer) ReloadSettings(filetype bool) {
+	settings := config.ParsedSettings()
+
+	if _, ok := b.LocalSettings["filetype"]; ok || filetype {
+		settings["filetype"] = b.Settings["filetype"]
+	} else if _, ok := settings["filetype"]; ok && settings["filetype"] != "unknown" {
+		b.Settings["filetype"] = settings["filetype"]
+	} else {
+		b.Settings["filetype"] = "unknown"
+	}
+	if b.Settings["filetype"] == "unknown" {
```

Comment:
`UpdateRules()` still supports `ft == ""` (equivalent to `ft == "unknown"`, undocumented). I'm not sure it makes sense to keep supporting it, but if we do, we should consistently handle it here as well?